framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -1293,7 +1294,8 @@ protected void addJoins(StringBuilder str, Collection<JoinBuilder<SearchCriteria
             fromIndex += onClause.length();
         }
 
-        str.delete(str.length() - 4, str.length());
+        int diff =  str.toString().trim().toLowerCase(Locale.ROOT).endsWith("where") ? 6 : 4;
```

Comment:
@Pearl1594 I'd leave out the trim() it is not clear why the 6 or 4 like this (or add a comment, but I prefer the code to explain itself)

Reply:
+1, add comment if code is not self-explanatory.